### PR TITLE
Set default env

### PR DESCRIPTION
**Pull request by ConnorRigby**
Every time I use ElixirMake, i end up needing to set most of these. I figured it would be easier to add it here.

### CODE REVIEW

**Comment by ConnorRigby** [comment]
Just now remembering `Mix.target()` is new.

**Comment by josevalim** [comment]
This looks great to me, we just need to document them. :+1: WDYT  @whatyouhide ?

File: lib/mix/tasks/compile.make.ex [hunk 1]
```diff
@@ -218,4 +219,34 @@ defmodule Mix.Tasks.Compile.ElixirMake do
...
+  # Returns a map of default environment variables
+  # Defauts may be overwritten.
+  defp default_env(config, env) do
+    root_dir = :code.root_dir()
+    erl_interface_dir = Path.join(root_dir, "usr")
+    [erts_dir] = Path.wildcard(Path.join(root_dir, "erts*"))
+
+    Map.merge(
+      %{
+        "MIX_TARGET" => to_string(Mix.target()),
```
Comment (josevalim) [inline lib/mix/tasks/compile.make.ex]: For both Mix.target you can read the env var directly or set a default: `System.get_env("MIX_TARGET") || "host"`.

**Comment by ConnorRigby** [comment]
Travis seems to be failing on `Mix.Project.consolidation_path/1`. I don't actually use this one i just added it for completeness. Thoughts?

**Comment by josevalim** [comment]
Let's remove it for now then. It was added only on v1.4 or something but unless we have a strong reason to break support for earlier Elixir versions, we probably shouldn't.

**Comment by josevalim** [comment]
@ConnorRigby the only remaining failure is about Mix.Project.manifest_path, which changed between Elixir versions. I think we have two options here:

1. Remove it
2. Use Mix.Project.manifest_path and friends in the tests

Thoughts?

**Comment by ConnorRigby** [comment]
I think removing it is fine. This one is another one I've never used personally, I just added it for completeness as well

**Comment by josevalim** [comment]
So let's go ahead and remove it and we should be good to go!

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**